tests/fixtures/nodes/src/nodes.rs

```diff
@@ -207,7 +207,7 @@ pub struct Nodes {
     stash_client: tokio::sync::Mutex<NillionChainClient>,
     next_payment_key_id: AtomicU64,
     next_signing_key_id: AtomicU64,
-    funded_payers: tokio::sync::Mutex<Vec<NillionChainClientPayer>>,
+    funded_payers: ManagedNillionChainClientPayer,
```

Comment:
What was the reasoning behind moving this logic to a new type? This pooling logic could have been here right? You essentially moved the existing `funded_payers` to a new type.